src/config/index.ts

```diff
@@ -175,7 +175,7 @@ export class Config extends (EventEmitter as {
 
 		return Bluebird.try(() => {
 			// Firstly validate all of the types as they are being set
-			this.validateConfigMap(keyValues);
+			keyValues = this.validateConfigMap(keyValues) as ConfigMap<T>;
```

Comment:
It'd be better to change `validateConfigMap` to specify it returns a `ConfigMap<T>`, that way it keeps the typing as close to the source as possible and less likely to become out of date/wrong